packages/decorators/src/watcher.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Add a watcher to watch for changes of a property on the specified object
+ *
+ * @param {Object} target prototype of the class
+ * @param {string} property name of the property to watch
+ * @param {Watcher} watcher function to invoke when the property gets a new value
+ */
+export const addWatcher = (
+    target: Object,
+    property: string,
+    watcher: Watcher
+) => {
+    const descriptor = Object.getOwnPropertyDescriptor(target, property);
+    const originalSetter = descriptor?.set;
+
+    Object.defineProperty(target, property, {
+        set: function (value) {
+            const oldValue = this[property];
+            if (originalSetter) {
+                originalSetter.apply(this, [value]);
+            }
+
+            watcher.apply(this, [value, oldValue]);
```

Comment:
here as well (feel free to ignore)

```suggestion
            watcher.call(this, value, oldValue);
```